toolchain/driver/compile_subcommand.cpp

```diff
@@ -820,6 +844,35 @@ auto CompileSubcommand::Run(DriverEnv& driver_env) -> DriverResult {
     return {.success = false};
   }
 
+  std::shared_ptr<clang::CompilerInvocation> clang_invocation;
+  // Build a clang invocation. We do this regardless of whether we're running
+  // check, because this is essentially performing further option validation,
+  // and we generally validate all options even if we're not using them for the
+  // selected phases of compilation.
+  // TODO: Share any arguments we specify here with the `carbon clang`
+  // subcommand.
+  {
+    llvm::SmallVector<std::string> clang_path_and_args = {
+        driver_env.installation->clang_path(),
+        // Propagate the target to Clang.
+        "-target",
+        options_.codegen_options.target.str(),
```

Comment:
Optional, but I always prefer joined options when we can...

```suggestion
        llvm::formatv("--target=$0", options_.codegen_options.target).str(),
```